src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/lazy_closure.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+/**
+ * @internal This class has been auto-generated by the Symfony Dependency Injection Component.
+ */
+class Symfony_DI_PhpDumper_Test_Lazy_Closure extends Container
+{
+    protected $parameters = [];
+    protected readonly \WeakReference $ref;
+
+    public function __construct()
+    {
+        $this->ref = \WeakReference::create($this);
+        $this->services = $this->privates = [];
+        $this->methodMap = [
+            'closure' => 'getClosureService',
+        ];
+
+        $this->aliases = [];
+    }
+
+    public function compile(): void
+    {
+        throw new LogicException('You cannot compile a dumped container that was already compiled.');
+    }
+
+    public function isCompiled(): bool
+    {
+        return true;
+    }
+
+    public function getRemovedIds(): array
+    {
+        return [
+            'foo' => true,
+        ];
+    }
+
+    protected function createProxy($class, \Closure $factory)
+    {
+        return $factory();
+    }
+
+    /**
+     * Gets the public 'closure' shared service.
+     *
+     * @return \Closure
+     */
+    protected static function getClosureService($container, $lazyLoad = true)
+    {
+        $containerRef = $container->ref;
+
+        return $container->services['closure'] = function (...$arguments) use ($containerRef) { static $service; return ($service ??= ($container = $containerRef->get()) ? new \Symfony\Component\DependencyInjection\Tests\Compiler\Foo() : null)->cloneFoo(...$arguments); };
```

Comment:
`null->cloneFoo(...$arguments)` will provide a totally WTF error. I think we need to have a much better handling of the case where the container ref is not valid anymore.

Reply:
This won't happen in practice. I wrote it this way to keep a onliner.